drivers/spi/spi_stm32lx.c

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) 2016 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <kernel.h>
+
+#include <board.h>
+#include <spi.h>
+#include <clock_control.h>
+
+#include <misc/util.h>
+
+#include <clock_control/stm32_clock_control.h>
+#include "spi_stm32lx.h"
+#include <drivers/spi/spi_stm32lx.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SPI_LEVEL
+#include <logging/sys_log.h>
+
+/* convenience defines */
+#define DEV_CFG(dev)							\
+	((const struct spi_stm32lx_config * const)(dev)->config->config_info)
+#define DEV_DATA(dev)							\
+	((struct spi_stm32lx_data * const)(dev)->driver_data)
+#define SPI_STRUCT(dev)							\
+	((volatile struct spi_stm32lx *)(DEV_CFG(dev))->base)
+
+
+#ifdef CONFIG_SPI_STM32LX_INTERRUPT
+static void spi_stm32lx_isr(void *arg)
+{
+	struct device * const dev = (struct device *)arg;
+
+	volatile struct spi_stm32lx *spi = SPI_STRUCT(dev);
+	struct spi_stm32lx_data *data = DEV_DATA(dev);
+	volatile u8_t *dr8 = &spi->dr.dr_as_u8;
+
+	if (spi->sr.bit.txe && data->current.tx_len) {
+		*dr8 = *data->current.tx_buf;
+		data->current.tx_buf++;
+		data->current.tx_len--;
+	}
+
+	if (spi->sr.bit.rxne && data->current.rx_len) {
+		*data->current.rx_buf = *dr8;
+		data->current.rx_buf++;
+		data->current.rx_len--;
+	}
+
+	if (!data->current.rx_len && !data->current.tx_len) {
+		spi->cr2.bit.rxneie = 0;
+		spi->cr2.bit.txeie = 0;
+
+		k_sem_give(&data->device_sync_sem);
+	}
+}
+#endif
+
+#define BR_SCALER_COUNT		8
+static const u32_t baud_rate_scaler[] = {
+	2, 4, 8, 16, 32, 64, 128, 256
+};
+
+/* Only support 8bit data size */
+#define DATA_SIZE_8_CFG		(3 + 4)
+
+static int spi_stm32lx_configure(struct device *dev, struct spi_config *config)
+{
+	volatile struct spi_stm32lx *spi = SPI_STRUCT(dev);
+	struct spi_stm32lx_data *data = DEV_DATA(dev);
+	const struct spi_stm32lx_config *cfg = DEV_CFG(dev);
+	u32_t flags = config->config;
+	u32_t clock;
+	int br;
+
+	clock_control_get_rate(data->clock,
+			(clock_control_subsys_t *)&cfg->pclken, &clock);
```

Comment:
parameters alignement:
foo(param1, ...
      paramn, ... );

Seems like you could break a line before &clock to get it

This rules applies everywhere on function declaration and call, please apply it where relevant, I see many places where it could be applied.